traitsui/editor.py

```diff
@@ -377,32 +392,86 @@ def get_undo_item(self, object, name, old_value, new_value):
         )
 
     def parse_extended_name(self, name):
-        """ Returns a tuple of the form ( context_object, 'name[.name...],
-            callable ) for a specified extended name of the form: 'name' or
-            'context_object_name.name[.name...]'.
+        """ Extract the object, name and a getter from an extended name
+
+        Parameters
+        ----------
+        name : str
+            The extended name to parse.
+
+        Returns
+        -------
+        object, name, getter : any, str, callable
+            The object from the context, the (extended) name of the
+            attributes holding the value, and a callable which gets the
+            current value from the context.
         """
-        col = name.find(".")
-        if col < 0:
+        parts = name.split(".", 1)
```

Comment:
Could also use `object_name, _, name = name.partition(".")` (and then test whether `name` is empty)

Reply:
I think `rpartition` is better given the subsequent code.